### PR TITLE
applyActionToAgent v2

### PR DESCRIPTION
**Pull request by Amix2**

### CODE REVIEW

File: simObj.py [hunk 1]
```diff
@@ -36,6 +36,9 @@ def GetCode(self, x: int, y: int) -> int:
 
 
 class Table(ISimObj):
+
+    required_agents = 1 # how many agents are required to move table
```
Comment (Yoanka) [inline simObj.py:40]: a TODO reminder to change this would be good.

File: document.py [hunk 2]
```diff
@@ -18,93 +18,153 @@ def __init__(self, sizeX: int, sizeY: int) -> None:
...
+        elif action.type == Action.move:
+            # move to a different square
+            if agent.grab:
+                actionRet = self.moveActionGrabbedAgent(agent, action.direction)
+            else:
+                actionRet = self.moveActionSoloAgent(agent, action.direction)
+
+        elif action.type == Action.grab:
+            actionRet = self.grabActionAgent(agent, action.direction)
+
```
Comment (Yoanka) [inline document.py:165]: also need actionRet = True for the wait action which is always valid

File: document.py [hunk 3]
```diff
@@ -18,93 +18,153 @@ def __init__(self, sizeX: int, sizeY: int) -> None:
...
+        targetPosition = agentPos + Action.directionToVector(direction)
+
+        # check if there's furniture there
+        target = self.GetCell(targetPosition[0], targetPosition[1])
+        targetCode = self.GetCellCode(targetPosition[0], targetPosition[1])
+        if not CellCodes.IsGrabbable(targetCode):
+            return False
+
+        # TODO: consider legal table-carrying configurations (can you only carry by holding short ends?)
+        agent.grab = direction
```
Comment (Yoanka) [inline document.py:136]: we might need to add a grab state to tables/chairs as well to keep track of how many agents are holding them and where they're trying to pull.

**Review by Yoanka — APPROVED** [review]

File: document.py (hunk 3 above)
Comment (Amix2) [reply document.py:136]: ok, good idea. It will require more care when grabbing / releasing, but will be useful. Ill add it to validation as well

**Comment by Amix2** [comment]
Updated

**Review by Yoanka — APPROVED** [review]

**Pull request merged**